### PR TITLE
feat: add C bindings

### PR DESCRIPTION
**Pull request by anonrig**

### CODE REVIEW

File: tests/ada_c.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+TEST(ada_c, getters) {
+  void* url = ada_parse(
+      "https://username:password@www.google.com:8080/"
+      "pathname?query=true#hash-exists");
+
+  ASSERT_TRUE(ada_is_valid(url));
+
+  // TODO: Fix ada_get_origin returning invalid address.
+  //  ASSERT_EQ(convert_string(ada_get_origin(url)),
```
Comment (anonrig) [inline tests/ada_c.cpp:27]: @lemire do you have any suggestions for this?

**Comment by anonrig** [comment]
By the way, should we expose the C API only through a build flag, or should we expose it always, like the current implementation does? @lemire

**Comment by anonrig** [comment]
@miguelteixeiraa I've added `ada_c.h` to the uploaded assets since we are adding C API to the release. Can you review?

**Comment by miguelteixeiraa** [comment]
Looks good!

**Pull request merged**